record/record.go

```diff
@@ -24,11 +24,12 @@
 //		records := record.NewReader(r)
 //		for {
 //			rec, err := records.Next()
-//			if err == io.EOF {
+//			if err == io.EOF || err == io.ErrUnexpectedEOF {
 //				break
 //			}
 //			if err != nil {
-//				return nil, err
+//				r.Recover()  // Skip corrupted records.
+//				continue
 //			}
 //			s, err := ioutil.ReadAll(rec)
 //			if err != nil {
```

Comment:
Do we also need to recover from this error? Is recovery going to be the same process for ReadAll as opposed to Next?